cmd/store.go

```diff
@@ -23,52 +24,46 @@ func (s *store) Get(IP string) (string, error) {
 	if role, ok := s.rolesByIP[IP]; ok {
 		return role, nil
 	}
+	log.Warnf("Using fallback role for IP %s", IP)
 	if s.defaultRole != "" {
-		log.Warnf("Using fallback role for IP %s", IP)
 		return s.defaultRole, nil
 	}
 	return "", fmt.Errorf("Unable to find role for IP %s", IP)
 }
 
 // OnAdd is called when a pod is added.
 func (s *store) OnAdd(obj interface{}) {
-	if pod, ok := obj.(*api.Pod); ok {
-		if pod.Status.PodIP != "" {
-			if role, ok := pod.Annotations[s.iamRoleKey]; ok {
-				s.mutex.Lock()
-				s.rolesByIP[pod.Status.PodIP] = role
-				s.mutex.Unlock()
-			}
+	pod := obj.(*api.Pod)
```

Comment:
With this change, it will panic which I don't think is the right thing to do. I agree that swallowing the error is not great but I'd prefer to log the error rather than panicking. Looking at a similar issue in the kubernetes project, they decided to log the error instead, see https://github.com/kubernetes/kubernetes/pull/18042/files